src/Netscript/NetscriptHelpers.tsx

```diff
@@ -297,17 +297,18 @@ function validateHGWOptions(ctx: NetscriptContext, opts: unknown): CompleteHGWOp
     return result;
   }
   if (typeof opts !== "object") {
-    throw errorMessage(ctx, `BasicHGWOptions must be an object if specified, was ${String(opts)}`);
+    // eslint-disable-next-line @typescript-eslint/no-base-to-string
+    throw errorMessage(ctx, `BasicHGWOptions must be an object if specified, was ${opts}`);
```

Comment:
Can we play nice with the linter?

The reason the rule no-base-to-string complains is because it is not obvious from the narrowed type that it has non-machine toString/toLocaleString (ie. one that doesn't return [object Object]).

Since in this case it was *not* an object and you expect one, the appropriate measure would be to JSON.stringify.

Reply:
I don't particularly disagree with the usage of JSON.stringify, but I'll only do that if d0sboots agrees. `${opts}` is only problematic when:
- `opts` is not an object.
- `opts`'s type is also not a built-in type with a nice toString implementation.
Off the top of my head, I cannot think of a normal use case.

Reply:
to be fair, opts _may be an object_. Object and object are not the same (emphasis: capitalization).

Something that is typeof !== 'object' (lowercase for emphasis) means that it is a non-null primitive. (null is also object).
- Symbol
- undefined
- primitive string (without new String())
- primitive number (without new Number())
- primitive boolean (without Boolean)
- functions

There is a loose check for null, so that removes undefined and null, only having to contend with the primitives that break the linter, which I believe is only Symbol.

To be exhaustive against the intended usage we could also check for Object.keys(opts).length in addition to typeof (strings are the only false positives on the keys check, the rest we have narrowed already).

The obvious culprit are Promises which notoriously simplify down to '{}' and can easily confuse a beginner.

Reply:
I'm not sure what you want to do here. As I said, `${opts}` is fine, maybe except some edge cases. If you mean there is a **practical** use case that we must replace `${opts}` with `${JSON.stringify(opts)}`, you have to point that out. The only thing that can fail is the Symbol, and Symbol is definitely not the practical use case.

Reply:
No change needed outside of suggestion for stringification. The rest is discussing alternatives.

Reply:
Agreed with catlover on two points:
* Denis, it was unclear to me what you were asking for (if anything). It seemed like you were asking for a change, but then you said not? It pays to be clear on this during a code review.
* I agree that JSON.stringify doesn't make sense here. It only works for a narrow set of types, and throws when it encounters other types. It is *not* something you want to be feeding `unknown` user data to in an error-handling scenario.